daily-patch-information/src/main/java/org/wso2/patchinformation/email/EmailContentCreator.java

```diff
@@ -0,0 +1,268 @@
+        emailBody += getStateTable(SECTION_HEADER_INACTIVE, COLUMN_NAMES_INACTIVE, "JIRA Create Date",
+                inactivePatches);
+        emailBody += getStateTable(SECTION_HEADER_SIGNING, COLUMN_NAMES, "Work Days In Signing",
+                patchesInSigning);
+        emailBody += getReleasedTable(jiraIssues);
+        emailBody += getSummeryTable(jiraIssues);
+        emailBody += EMAIL_FOOTER;
+        return emailBody;
+    }
+
+    /**
+     * Builds the html string for the Summary table.
+     *
+     * @param jiraIssues ArrayList of all JIRA issues
+     * @return the html code for the table
+     */
+    private String getSummeryTable(ArrayList<JIRAIssue> jiraIssues) {
+
+        String table = SECTION_HEADER_SUMMARY;
+        table += COLUMN_NAMES_SUMMARY;
+        jiraIssues.sort(new ReportDateComparator());
+        ArrayList<HtmlTableRow> jirasToHtml = new ArrayList<>(jiraIssues);
+        table += getTableRows(jirasToHtml);
+        table += "</table>";
+        return table;
+    }
+
+    /**
+     * Returns an ArrayList of all inactive patches.
+     *
+     * @param jiraIssues ArrayList of JIRA issues
+     * @return all patches
+     */
+    private ArrayList<InactivePatch> getAllInactivePatches(ArrayList<JIRAIssue> jiraIssues) {
+
+        ArrayList<InactivePatch> inactivePatches = new ArrayList<>();
+        for (JIRAIssue jiraIssue : jiraIssues) {
+            if (jiraIssue.getOpenPatchesInJIRA().isEmpty()) {
+                inactivePatches.addAll(jiraIssue.getInactivePatches());
+            }
+        }
+        inactivePatches.sort(new JIRADateComparator());
+        return inactivePatches;
+    }
+
+    /**
+     * Returns an ArrayList of all open patches.
+     *
+     * @param jiraIssues ArrayList of JIRA issues
+     * @return all open patches
+     */
+    private ArrayList<OpenPatch> getAllOpenPatches(ArrayList<JIRAIssue> jiraIssues) {
+
+        ArrayList<OpenPatch> openPatches = new ArrayList<>();
+        for (JIRAIssue jiraIssue : jiraIssues) {
+            openPatches.addAll(jiraIssue.getOpenPatchesInJIRA());
+        }
+        return openPatches;
+    }
+
+    /**
+     * Assigns patches to States.
+     *
+     * @param openPatches ArrayList of all patches to be recorded.
+     */
+    private void assignPatchesToStates(ArrayList<Patch> openPatches, ArrayList<Patch> patchesInSigning,
+                                       ArrayList<Patch> patchesInDevelopment) {
+
+        for (Patch openPatch : openPatches) {
+            switch (openPatch.getState()) {
+                case IN_DEV:
+                    patchesInDevelopment.add(openPatch);
+                    break;
+                case IN_PATCH_QUEUE:
+                    patchesInDevelopment.add(openPatch);
+                    break;
+                case IN_SIGNING:
+                    patchesInSigning.add(openPatch);
+                    break;
+            }
+        }
+        patchesInDevelopment.sort(new DaysInStateComparator());
```

Comment:
You don't need to pass comparator if the original Patch class had implemented the comparator as mentioned in the above comments.